backend/util/FileUploadController.js

```diff
@@ -0,0 +1,154 @@
+  }
+};
+
+const DeleteFile = async (req, res) => {
+  const ID = req.params.ID;
+  File.findById(ID)
+    .then(async (file) => {
+      try {
+        await deleteFileAWS(file.S3_ID);
+      } catch (e) {
+        console.log(e);
+        res.status(500).json({ error: "Internal Error. Could not upload file to AWS" });
+      }
+      await File.deleteOne({ _id: ID });
+      res.send("File successfully deleted");
+    })
+    .catch((e) => {
+      console.log(e);
+      res.status(400).json({ error: "Error, file with provided id does not exist" });
+    });
+};
+
+const UpdateFile = async (req, res) => {
+  try {
+    const ID = req.params.ID;
+    const updated_file = req.file;
+    File.findById(ID)
+      .then(async (fileobj) => {
+        if (req.file && req.body.updated_file_name) {
+          try {
+            await deleteFileAWS(fileobj.S3_ID);
+            const result = await uploadFile(updated_file);
+            Object.assign(fileobj, { S3_ID: result.key, name: req.body.updated_file_name });
+            fileobj.save();
+          } catch (e) {
+            console.log(e);
+            res.status(400).json({ error: "Cannot update file details" });
+          }
+        } else if (req.file) {
+          try {
+            await deleteFileAWS(fileobj.S3_ID);
+            const result = await uploadFile(updated_file);
+            Object.assign(fileobj, { S3_ID: result.key }); // so that UpdatedAt changes
+            fileobj.save();
+          } catch (e) {
+            console.log(e);
+            res.status(400).json({ error: "Cannot update file" });
+          }
+        } else if (req.body.updated_file_name) {
+          try {
+            Object.assign(fileobj, { name: req.body.updated_file_name });
+            fileobj.save();
+          } catch (e) {
+            console.log(e);
+            res.status(400).json({ error: "Cannot update file name" });
+          }
+        } else {
+          res.status(400).json({ error: "please enter an input" });
+        }
+      })
+      .catch((e) => {
+        console.log(e);
+        res.status(500).json({ error: "Error, Old file does not exist, please uplode new" });
+      });
+  } catch (e) {
+    console.log(e);
+    res.status(400).json({ error: "Error, cannot updated file" });
+  }
+};
+
+const DisplayFile = async (req, res) => {
+  try {
+    const ID = req.params.ID;
+    File.findById(ID)
+      .then(async (file) => {
+        const readStream = getFileStream(file.S3_ID);
+        readStream.pipe(res);
+      })
+      .catch(() => {
+        res.status(500).json({ error: "Error, file does not exist" });
+      });
+  } catch (e) {
+    console.log(e);
+    res.status(400).json({ error: "Error, cannot display file" });
+  }
+};
+
+const DeleteCategory = async (req, res) => {
+  try {
+    const category_id = req.params.category_id;
+    await File.find({ Category_ID: category_id })
+      .then(async (files) => {
+        files.forEach(async (file) => {
```

Comment:
If anything goes wrong with deleting a file on AWS, this won't fail the way we want it to.  Instead of using `files.forEach()`, something like this will do what we want:

```js
Promise.all(files.map(async (file) => {
    await deleteFileAWS(file.S3_ID);
    await File.deleteOne({ name: file.name });
})).catch(() => res.status(500).json({ error: "Failed to delete files" });
```